src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseBase.cpp

```diff
@@ -2390,21 +2377,22 @@ void CompositionalMultiphaseBase::saveConvergedState( ElementSubRegionBase & sub
   arrayView2d< real64, compflow::USD_COMP > const & compDens_n =
     subRegion.template getField< fields::flow::globalCompDensity_n >();
   compDens_n.setValues< parallelDevicePolicy<> >( compDens );
+  if( m_isFixedStressPoromechanicsUpdate )
+  {
+    arrayView2d< real64, compflow::USD_COMP > const & compDens_k =
+      subRegion.template getField< fields::flow::globalCompDensity_k >();
+    compDens_k.setValues< parallelDevicePolicy<> >( compDens );
+  }
 }
 
-void CompositionalMultiphaseBase::saveIterationState( DomainPartition & domain ) const
+void CompositionalMultiphaseBase::saveSequentialIterationState( DomainPartition & domain ) const
 {
-  FlowSolverBase::saveIterationState( domain );
+  FlowSolverBase::saveSequentialIterationState( domain );
 }
 
-void CompositionalMultiphaseBase::saveIterationState( ElementSubRegionBase & subRegion ) const
+void CompositionalMultiphaseBase::saveSequentialIterationState( ElementSubRegionBase & subRegion ) const
 {
-  FlowSolverBase::saveIterationState( subRegion );
-
-  if( !subRegion.hasField< fields::flow::globalCompDensity_k >() )
-  {
-    return;
-  }
+  FlowSolverBase::saveSequentialIterationState( subRegion );
 
   arrayView2d< real64 const, compflow::USD_COMP > const compDens = subRegion.template getField< fields::flow::globalCompDensity >();
   arrayView2d< real64, compflow::USD_COMP > const compDens_k = subRegion.template getField< fields::flow::globalCompDensity_k >();
```

Comment:
is this the valued at outer iteration `k`?

Reply:
yes, same logic as pressure_k